inference-engine/tests/unit/vpu/middleend_tests/passes_tests/adjust_data_batch_tests.cpp

```diff
@@ -0,0 +1,389 @@
+
+    ASSERT_EQ(data11, testModel.getOutputs());
+}
+
+TEST_F(VPU_AdjustDataBatchTest_New_Inf, Case4_BranchedWithBatchItemsInTheEnd) {
+    //                                                                                      -> (Batched) -> [Output]
+    // [Input] -> (Split) -> (Split) -> (Split) -> (Split) -> (Split) -> (Split) -> (Batch)
+    //                                                                                      -> (Batched) -> [Output]
+
+    testModel.createInputs(1);
+    testModel.createOutputs(2);
+
+    testModel._addStage({InputInfo::fromNetwork()}, {OutputInfo::intermediate()});
+    testModel.setStageBatchInfo(0, {{0, BatchSupport::Split}});
+
+    testModel._addStage({InputInfo::fromPrevStage(0)}, {OutputInfo::intermediate()});
+    testModel.setStageBatchInfo(1, {{0, BatchSupport::Split}});
+
+    testModel._addStage({InputInfo::fromPrevStage(1)}, {OutputInfo::intermediate()});
+    testModel.setStageBatchInfo(2, {{0, BatchSupport::Split}});
+
+    testModel._addStage({InputInfo::fromPrevStage(2)}, {OutputInfo::intermediate()});
+    testModel.setStageBatchInfo(3, {{0, BatchSupport::Split}});
+
+    testModel._addStage({InputInfo::fromPrevStage(3)}, {OutputInfo::intermediate()});
+    testModel.setStageBatchInfo(4, {{0, BatchSupport::Split}});
+
+    testModel._addStage({InputInfo::fromPrevStage(4)}, {OutputInfo::intermediate()});
+    testModel.setStageBatchInfo(5, {{0, BatchSupport::Split}});
+
+    testModel._addStage({InputInfo::fromPrevStage(5)}, {OutputInfo::intermediate()});
+    testModel._addStage({InputInfo::fromPrevStage(6)}, {OutputInfo::fromNetwork(0)});
+    testModel._addStage({InputInfo::fromPrevStage(6)}, {OutputInfo::fromNetwork(1)});
+
+    RunPass();
+
+    const auto& data0 = singleElement(checkSingleLoopStart(testModel.getInputs().at(0)));
+    const auto& data1 = CheckSingleConnection(data0, 0);
+    const auto& data2 = CheckSingleConnection(data1, 1);
+    const auto& data3 = CheckSingleConnection(data2, 2);
+    const auto& data4 = CheckSingleConnection(data3, 3);
+    const auto& data5 = CheckSingleConnection(data4, 4);
+    const auto& data6 = CheckSingleConnection(data5, 5);
+    const auto& data7 = singleElement(checkSingleLoopEnd(data6));
+
+    const auto& data8 = CheckSingleConnection(data7, 6, batchSize);
+
+    const auto& branches = checkBranches(data8, {StageType::None, StageType::None});
+    const auto& withBatch = branches[0];
+    const auto& withBatch_1 = branches[1];
+
+    ASSERT_EQ(withBatch->producer()->attrs().get<int>("test_ind"), 7);
+    ASSERT_EQ(withBatch->desc().dim(Dim::N), batchSize);
+    ASSERT_EQ(withBatch, testModel.getOutputs().at(0));
+
+    ASSERT_EQ(withBatch_1->producer()->attrs().get<int>("test_ind"), 8);
+    ASSERT_EQ(withBatch_1->desc().dim(Dim::N), batchSize);
+    ASSERT_EQ(withBatch_1, testModel.getOutputs().at(1));
+}
+
+TEST_F(VPU_AdjustDataBatchTest_New_Inf, DISABLED_Case5_SubGraphsWithSingleSplitInput) {
```

Comment:
why did you disable the test?

Reply:
Because this test was created to test functionality that `master` does not yet have (it is in the `release` branch)

Reply:
ok. Could you create a ticket to undisable the test after some merge?

Reply:
Sure.